guides/leon/debain-email.md

```diff
@@ -0,0 +1,308 @@
+   
+2. Select options as follows in the installation (pretty much a standard setup)
+    1. Choose Graphical install
+    2. English
+    3. United States
+    4. American English
+    5. "debian"
+    6. "debian-email.com"
+    7. {super secret password}
+    8. {some user}
+    9. {some user password}
+    10. Eastern
+    11. Guided - sue entire disk
+    12. All files in one partition
+    13. Finish partitioning
+    14. Yes - write to file
+    15. No
+    16. United States
+    17. ftp.us.debian.org
+    18. No - no popularity contest
+    19. For software selection pick only: standard system utilities
+    20. Yes - to grub
+    21. /dev/sda - should be the only disk
+4. Now you should see a terminal login prompt. Login.
+![login](login.PNG)
+<br>
+
+<a id="postfix"></a>
+## Install Postfix 
+Here we will install Postfix.
+
+1. Preperations
+
+    First become superuser and fetch updates
+    ``` bash
+    su
+    # Enter the {super secret password}
+    apt-get update
+    apt-get upgrade
+    ```
+    
+    Now install some packages that will be useful later
+    ```bash
+    apt-get install -y net-tools wget curl
+    ```
+    
+    Now open up /etc/host.conf file with `vi` configure it so it looks like so.
+    ```bash
+    order hosts, bind
+    multi on
+    ```
+    
+    Now we add our ip and domain name 'debian-email.com' to the host file so it will internally resolve. Make sure someone on the DNS server has debian-email.com point to this box - or add it manually to each windows/linux host file.
+    
+    First, get the IP by running `ifconfig` . The info we want will appear like so  `inet 10.0.2.15` . Now run these two commands, use your IP instead of 10.0.2.15
+    ```bash
+    hostnamectl set-hostname mail.debian-email.com
+    echo "10.0.2.15" debian-email.com mail.debian-email.com >> /etc/hosts
+    ```
+    
+    Now reboot: `reboot` . Then become superuser again
+
+2.
+    Install postfix
+    
+    ``` bash
+    apt-get install -y postfix
+    # A graphical-esc inerface pops up
+    # For Mail Configuration: Internet Site
+    # Our mail name will be: "debian-email.com"
+    ```
+3. No we need to make some adjustments to the postfix configuration
+    
+    Go to `/etc/postfix` and edit the `main.cf` file with `vi`
+    ```bash
+    # Find the line that reads myhostname= and change to
+    myhostname=mail.debian-email.com
+    
+    # Add the following line
+    mydomain=debian-email.com
+    home_mailbox = Maildir/
+    smtpd_sasl_type = dovecot
+    smtpd_sasl_path = private/auth
+    smtpd_sasl_auth_enable = yes
+    smtpd_sasl_security_options = noanonymous
+    smtpd_sasl_local_domain = $myhostname
+    smtpd_recipient_restrictions = permit_mynetworks,permit_auth_destination,permit_sasl_authenticated,reject
+    ```

```

Comment:
If you know what those options mean, you should explain each one. It's okay if you don't.